### PR TITLE
feat(split): add size and on-update:size, controlled properties

### PR DESCRIPTION
**Pull request by XueHua-s**

### CODE REVIEW

**Comment by XueHua-s** [comment]
feat(split): add size and on-update:size, controlled properties

**Comment by XueHua-s** [comment]
更改了一些，自己文档写错的地方

**Comment by XueHua-s** [comment]
增加, 使用受控值的案例

**Comment by XueHua-s** [comment]
修复最大最小值的问题.

File: src/version.ts [hunk 1]
```diff
@@ -1 +1 @@
-export default '2.37.3'
+export default '2.37.4'
```
Comment (OrbisK) [inline src/version.ts:1]: please dont update this in PR

File: src/split/src/Split.tsx [hunk 2]
```diff
@@ -28,6 +29,10 @@ export const splitProps = {
     type: Number,
     default: 0.5
   },
+  size: {
+    type: Number,
+    default: 0
```
Comment (OrbisK) [inline src/split/src/Split.tsx:34]: not sure if this should default to 0. this should default to undefined so default-size will work and 0 as value should still be possible

File: CHANGELOG.en-US.md [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 ## NEXT_VERSION
 
+### Feat
+
+-Feat `n-split`'s `size` and `on-update:size` add size and on-update:size, controlled properties.
```
Comment (OrbisK) [inline CHANGELOG.en-US.md:7]:
```suggestion
- Add `n-split` `size` prop and `on-update:size` event.
```

**Review by OrbisK — CHANGES_REQUESTED** [review]
Thank you for your contribution 🚀

**Comment by XueHua-s** [comment]
The changes have been made

File: src/split/demos/enUS/index.demo-entry.md [hunk 4]
```diff
@@ -21,6 +21,8 @@ slot.vue
 | Name | Type | Default | Description | Version |
 | --- | --- | --- | --- | --- |
 | default-size | `number` | `0.5` | Default split size, 0-1 is a percentage. | 2.36.0 |
+| size | `number` | `undefined` | Split is the controlled split size, with 0-1 representing the percentage | 2.37.4 |
+| on-update:size | `(value: number) => void` | `undefined` | When a controlled property is used, it is triggered when the size property of the component changes | 2.37.4 |
```
Comment (OrbisK) [inline src/split/demos/enUS/index.demo-entry.md:25]:
```suggestion
| size | `number` | `undefined` | Split is the controlled split size, with 0-1 representing the percentage | NEXT_VERSION |
| on-update:size | `(value: number) => void` | `undefined` | When a controlled property is used, it is triggered when the size property of the component changes | NEXT_VERSION |
```
see https://github.com/tusen-ai/naive-ui/blob/main/CONTRIBUTING.zh-CN.md#关于文档和变更日志的格式

File: src/split/demos/zhCN/index.demo-entry.md [hunk 5]
```diff
@@ -21,6 +22,8 @@ slot.vue
 | 名称 | 类型 | 默认值 | 说明 | 版本 |
 | --- | --- | --- | --- | --- |
 | default-size | `number` | `0.5` | Split 的默认分割大小，0-1 代表百分比 | 2.36.0 |
+| size | `number` | `undefined` | Split 的受控分割大小，0-1 代表百分比 | 2.37.4 |
+| on-update:size | `(value: number) => void` | `undefined` | 使用受控属性时, 组件size属性变化时触发 | 2.37.4 |
```
Comment (OrbisK) [inline src/split/demos/zhCN/index.demo-entry.md:26]:
```suggestion
| size | `number` | `undefined` | Split 的受控分割大小，0-1 代表百分比 | NEXT_VERSION |
| on-update:size | `(value: number) => void` | `undefined` | 使用受控属性时, 组件size属性变化时触发 | NEXT_VERSION |
```

File: src/split/src/Split.tsx [hunk 6]
```diff
@@ -72,8 +76,23 @@ export default defineComponent({
...
+            currentSize.value = newSize
+          }
+        }
+      )
+      watch(
+        () => currentSize.value,
+        (newSize) => {
+          emit('update:size', newSize)
+        }
+      )
```
Comment (OrbisK) [inline src/split/src/Split.tsx:94]: if props.size changes it will trigger currentSize and will trigger @update:size.

**Review by OrbisK — CHANGES_REQUESTED** [review]

File: src/split/src/Split.tsx (hunk 6 above)
Comment (OrbisK) [reply src/split/src/Split.tsx:94]: i guess you can copy paste some code from checkbox for default and prop merging:
https://github.com/tusen-ai/naive-ui/blob/fce7b64ead4ba18d26b0115dc500303fdd70b3fc/src/checkbox/src/Checkbox.tsx#L153

**Comment by OrbisK** [comment]
> The changes have been made

Thank you for your additonal work. There are some small adjustments left todo.

**Comment by XueHua-s** [comment]
I have used the merge method with reference to the code in another file.

**Comment by liuyang0826** [comment]
最好是在props增加onUpdateSize和onUpdat:size事件，和其他组件保持一致

**Comment by XueHua-s** [comment]
> @XueHua-s 
Added type onUpdate:size

**Comment by XueHua-s** [comment]
> @XueHua-s 最好是在props增加onUpdateSize和onUpdat:size事件，和其他组件保持一致

Added type onUpdate:size

**Comment by XueHua-s** [comment]
Added type onUpdate:size

**Comment by 07akioni** [comment]
Basically good. For needed changes, you can see what I've done in 7cbaeaa4b

**Pull request merged**